op-e2e/system_test.go

```diff
@@ -646,6 +649,132 @@ func TestSystemMockP2P(t *testing.T) {
 	require.Contains(t, received, receiptVerif.BlockHash)
 }
 
+// TestSystemMockPeerScoring sets up a L1 Geth node, a rollup node, and a L2 geth node and then confirms that
+// the nodes can sync L2 blocks before they are confirmed on L1.
+func TestSystemMockPeerScoring(t *testing.T) {
```

Comment:
Ideally we'd also have a test where one node is sending invalid messages and check it gets disconnected.  Might be reasonable to do by having one of the nodes running a sequencer with the wrong key since all the published blocks would then be considered invalid gossip.

Reply:
Nice! I can add that!

Reply:
+1, can also make one of the receiving verifier nodes here bad, and test that the good verifier nodes continue to receive payloads, while the bad one should stop getting them & be disconnected.